### PR TITLE
expression: revert pr 6621

### PR DESCRIPTION
**Pull request by zhexuany**
https://github.com/pingcap/tidb/pull/6621 is not correct way to do it. Mysql use `datetime` to compare. The real reason is that we did not handle daylight saving time.

### CODE REVIEW

**Comment by zhexuany** [comment]
@XuHuaiyu @coocood PTAL

File: expression/integration_test.go [hunk 1]
```diff
@@ -2721,12 +2721,6 @@ func (s *testIntegrationSuite) TestCompareBuiltin(c *C) {
 	tk.MustExec("create table t(a bigint unsigned)")
 	tk.MustExec("insert into t value(17666000000000000000)")
 	tk.MustQuery("select * from t where a = 17666000000000000000").Check(testkit.Rows("17666000000000000000"))
-
-	tk.MustExec("drop table if exists t")
```
Comment (tiancaiamao) [inline expression/integration_test.go]: comment it and leave a TODO ?

Comment (shenli) [reply expression/integration_test.go]: We already have an issue about it. So re-open the issue is enough.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
Do not forget to backport this to the release-2.0.

**Pull request merged**